pkg/vsphere/vm/vm.go

```diff
@@ -611,6 +614,23 @@ func (vm *VirtualMachine) DisableDestroy(ctx context.Context) {
 
 	err := m.DisableMethods(ctx, obj, method, "VIC")
 	if err != nil {
-		log.Warnf("failed to disable method %s for %s: %s", method[0].Method, obj[0], err)
+		log.Warnf("Failed to disable method %s for %s: %s", method[0].Method, obj[0], err)
+	}
+}
+
+// EnableDestroy attempts to enable the VirtualMachine.Destroy_Task method
+// so that the VC UI can enable the VM "delete" action.
+func (vm *VirtualMachine) EnableDestroy(ctx context.Context) {
```

Comment:
Let's keep it like this for now. I agree with your point. Here I just try to be consistent with the way the other similar method `DisableDestroy()` deals with its own error.